### PR TITLE
Wav2vec2

### PR DESCRIPTION
**Pull request by DmitryMalishev**
Added demo-notebook 'wav2vec2-quantize'.
To launch the notebook openvino-dev ver. 22.1+ is expected, not working with earlier versions.

### CODE REVIEW

**Comment by ryanloney** [comment]
@DmitryMalishev that you for submitting this PR. We will start reviewing it soon, but can you add one additional step to run inference, decoding and print output for a sample audio file?

You can see how we show the ASR output in a similar notebook: https://github.com/openvinotoolkit/openvino_notebooks/blob/main/notebooks/211-speech-to-text/211-speech-to-text.ipynb

File: notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb
Comment (KodiaqQ) [inline notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb:259]: Also, it's better to use `self.config` after `super()` calling.

File: notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb
Comment (KodiaqQ) [inline notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb:274]: If I were you, I'd rather return a dictionary (`{input_name: input_data}`) instead of NumPy arrays, to avoid the issues with the data shapes and etc.
@AlexKoff88, what do you think? Should it be better to use a dictionary here?

File: notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb
Comment (KodiaqQ) [inline notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb:342]: Do we really need to define `value()` here? It's needed only for the Accuracy Aware quantization algorithm.

File: notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb
Comment (KodiaqQ) [inline notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb:469]: Where do layers like `Add_843`, `Conv_19`, etc. come from? The FP IR doesn't contain these layers. Please, verify that all layers from the ignored scope exist in the model. Also, `214` in the ignored scope is a WA for the known runtime issue. Please, note, that it should be removed in the future.

File: notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb
Comment (KodiaqQ) [inline notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb:456]: How is this configuration was found for this model? Is it gives us a more accurate model in comparison with the default configuration?

**Comment by DmitryMalishev** [comment]
> @DmitryMalishev that you for submitting this PR. We will start reviewing it soon, but can you add one additional step to run inference, decoding and print output for a sample audio file?
> 
> You can see how we show the ASR output in a similar notebook: https://github.com/openvinotoolkit/openvino_notebooks/blob/main/notebooks/211-speech-to-text/211-speech-to-text.ipynb

@ryanloney, do you mean to add the following at the end of the notebook: [1] take one audio from the dataset [2] play it back  [3] inference with _quantized_ only model [4] print the recognized text ?

File: notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb
Comment (DmitryMalishev) [reply notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb:256]: @KodiaqQ, demo-notebooks #105 and #113 contain this useless type checking, I only left it for consistency. Since you noticed it, I'm considering to remove this check.

File: notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb
Comment (DmitryMalishev) [reply notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb:259]: Good point, thank you.

File: notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb
Comment (DmitryMalishev) [reply notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb:342]: Unfortunately, we need, API makes is mandatory: https://github.com/openvinotoolkit/openvino/blob/master/tools/pot/openvino/tools/pot/api/metric.py

File: notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb
Comment (DmitryMalishev) [reply notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb:456]: Particular layers are not currently processed automatically during quantization which end up in dramatical accuracy drop, so they are ignored for now as a work-around.

File: notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb
Comment (DmitryMalishev) [reply notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb]: You're right, ignored layers block is useless. I've just submitted fix. Only '214' left as a WA for known issue.

File: notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb
Comment (DmitryMalishev) [reply notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb:454]: 'activation' section is required for now, significant accuracy reduction without it. '214' layer ignore is WA for known issue. That's it, the rest is default.

**Comment by ryanloney** [comment]
> > @DmitryMalishev that you for submitting this PR. We will start reviewing it soon, but can you add one additional step to run inference, decoding and print output for a sample audio file?
> > You can see how we show the ASR output in a similar notebook: https://github.com/openvinotoolkit/openvino_notebooks/blob/main/notebooks/211-speech-to-text/211-speech-to-text.ipynb
> 
> @ryanloney, do you mean to add the following at the end of the notebook: [1] take one audio from the dataset [2] play it back [3] inference with _quantized_ only model [4] print the recognized text ?

@DmitryMalishev yes. If we use this notebook for a webinar, workshop or developer training program, the presenter will need to show sample input and output to the quantized model.

File: notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb
Comment (AlexKoff88) [reply notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb:272]: It should, especially if to note the model is dynamic in terms of the size of input data so the dict should be safer as we convinced before.

File: notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb
Comment (AlexKoff88) [inline notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb:47]: If the intent is to use OpenVINO 2022.1 anyway, I would highly recommend using a new importing system for POT. You can find an example [here](https://github.com/openvinotoolkit/openvino/blob/master/tools/pot/openvino/tools/pot/api/samples/classification/classification_sample.py).

File: notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb
Comment (AlexKoff88) [inline notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb:191]: Do we need all these MO parameters, like "--input", "--input_shape", "--data_type"? Can we minimize them and keep only required, like path to model and output folder?

**Comment by AlexKoff88** [comment]
Overall it looks good to me. Thanks, @DmitryMalishev!

**Comment by DmitryMalishev** [comment]
@ryanloney @KodiaqQ @AlexKoff88 Thanks for you comments, I took into account all of them and just submitted new version with changes.

Please note, Inference Example won't work until the dynamic shape issue isn't fixed.

**Comment by ryanloney** [comment]
Thanks @DmitryMalishev. Is this the dynamic shape issue? 
`ValueError: get_shape was called on a descriptor::Tensor with dynamic shape` (at step 6)

Is there a solution? Or do we need to wait until a future release like 2022.2?

**Comment by DmitryMalishev** [comment]
> Thanks @DmitryMalishev. Is this the dynamic shape issue? `ValueError: get_shape was called on a descriptor::Tensor with dynamic shape` (at step 6)
> 
> Is there a solution? Or do we need to wait until a future release like 2022.2?

@ryanloney,
The error you've faced is a part of the dynamic shape issue. This part is already fixed and available through manual installing OpenVino from the github's head. By doing this you can go further with the notebook.

Another part affects making prediction at the Model Usage Example section. Fixing this part is pending and not available yet.

The release with both fixes and thus fully workable notebook will be announced once it becomes available.

**Comment by DmitryMalishev** [comment]
@paularamo @ryanloney This notebook is ready for merge, no other changes are planned. Could you please review and merge?

**Review by KodiaqQ — APPROVED** [review]
LGTM, the main functionality is shown and functional (excluding the moment with WA).

**Review by AlexKoff88 — APPROVED** [review]

File: notebooks/107-speech-recognition-quantize-wav2vec2/107-speech-recognition-quantize-wav2vec2.ipynb
Comment (Debskij) [inline notebooks/107-speech-recognition-quantize-wav2vec2/107-speech-recognition-quantize-wav2vec2.ipynb:591]: Please use single line inference
`compiled_model(feed_dict)[output_layer]`

**Review by Debskij — CHANGES_REQUESTED** [review]

File: notebooks/107-speech-recognition-quantize-wav2vec2/107-speech-recognition-quantize-wav2vec2.ipynb
Comment (DmitryMalishev) [reply notebooks/107-speech-recognition-quantize-wav2vec2/107-speech-recognition-quantize-wav2vec2.ipynb]: @Debskij thanks for the comment. I changed the inference according to the documentation:
https://docs.openvino.ai/latest/notebooks/002-openvino-api-with-output.html

File: notebooks/107-speech-recognition-quantize-wav2vec2/107-speech-recognition-quantize-wav2vec2.ipynb
Comment (Debskij) [inline notebooks/107-speech-recognition-quantize-wav2vec2/107-speech-recognition-quantize-wav2vec2.ipynb:593]:
```suggestion
    "output_layer = compiled_model.output(0)"
```

Comment (DmitryMalishev) [reply notebooks/107-speech-recognition-quantize-wav2vec2/107-speech-recognition-quantize-wav2vec2.ipynb]: Thanks, this simplification also looks good. I submitted the commit, it seems to use square brackets, not round ones.

**Review by ryanloney — APPROVED** [review]
@DmitryMalishev please change `notebooks/107-speech-recognition-quantize-wav2vec2` directory to `notebooks/107-speech-recognition-quantization`

And please update README.md in root level to include: 
```
| [107-speech-recognition-quantization](notebooks/107-speech-recognition-quantization/) | Optimize and quantize a pre-trained Wav2Vec2 speech model |
```
Currently it is only in `notebooks/README.md` and not main README

After that, we can merge it @paularamo / @adrianboguszewski

**Comment by DmitryMalishev** [comment]
@ryanloney thanks for corrections, I updated directory structure and root 'readme'
@paularamo @adrianboguszewski please launch checking workflows

**Comment by DmitryMalishev** [comment]
Failures in recent nbval checks are caused by notebook-105 and not relevant to current notebook.

**Review by Debskij — APPROVED** [review]

**Comment by adrianboguszewski** [comment]
@ryanloney Cannot merge it due to random CI errors and too strict merging rules or missing admin permissions.

**Pull request merged**
